neard/src/cli.rs

```diff
@@ -441,6 +446,47 @@ impl RunCmd {
     }
 }
 
+async fn signal_handlers(home_dir: &Path, rx_crash: Receiver<()>) -> &str {
+    use tokio::signal::unix::{signal, SignalKind};
+    let rx_int = async_stream::stream! {
+        let mut sigint = signal(SignalKind::interrupt()).unwrap();
+            sigint.recv().await;
+            yield "SIGINT";
+    };
+    let rx_hup = async_stream::stream! {
+        let mut sighup = signal(SignalKind::hangup()).unwrap();
+        loop {
+            sighup.recv().await;
+            yield "SIGHUP";
+        }
+    };
+    let rx_term = async_stream::stream! {
+        let mut sigterm = signal(SignalKind::terminate()).unwrap();
+            sigterm.recv().await;
+            yield "SIGTERM";
+    };
+    let rx_crash = async_stream::stream! {
+        let _ = rx_crash.fuse().await;
+        yield "Client died";
+    };
+    pin_mut!(rx_int);
+    pin_mut!(rx_hup);
+    pin_mut!(rx_term);
+    pin_mut!(rx_crash);
+
+    let watched_path = home_dir.join("log_config.json");
+    let log_config_watcher = LogConfigWatcher { watched_path };
```

Comment:
I cannot help but think that we should read that config if it exists at node startup.  Maybe have `EnvFilterBuilder` handle creation of the watcher and return `(env_filter, log_config_watcher)` in `finish` method?